controllers/capabilities/backend_controller.go

```diff
@@ -74,12 +76,57 @@ func (r *BackendReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		reqLogger.V(1).Info(string(jsonData))
 	}
 
+	// Ignore deleted Backends, this can happen when foregroundDeletion is enabled
+	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
+	if backend.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(backend, backendFinalizer) {
+		err = controllerhelper.ReconcileFinalizers(backend, r.Client(), backendFinalizer)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		err = r.removeBackend(backend)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	}
+
 	// Ignore deleted Backends, this can happen when foregroundDeletion is enabled
 	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
 	if backend.DeletionTimestamp != nil {
 		return ctrl.Result{}, nil
 	}
 
+	if !controllerutil.ContainsFinalizer(backend, backendFinalizer) {
+		err = controllerhelper.ReconcileFinalizers(backend, r.Client(), backendFinalizer)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if backend.Spec.ProviderAccountRef != nil {
```

Comment:
I've added it in because it seems that the spec.ProviderAccountRef is not a required field, so it might be missing. Unless I'm reading it wrong, will have a look again. Thank you